field/field.go

```diff
@@ -35,6 +38,20 @@ func (f Static) Equal(to Static) bool {
 	return true
 }
 
+func (f Static) IsRoot() bool {
+	return len(f) == 0
+}
+
+func (f Static) IsParent(child Static) bool {
```

Comment:
The meaning of "parent" in this context isn't obvious. I think the intent would be clearer if it were remade as HasPrefix, with behavior analogous to bytes.HasPrefix and strings.HasPrefix.
```go
func (f Static) HasPrefix(prefix Static) bool {
	return len(f) >= len(prefix) && prefix.Equal(f[:len(prefix)])
}

Reply:
I made this change in af793d6fa5b91ff2265691ec82fcb32a68fb1841. The changed semantics make the client code a bit more complicated. What do you think about adding 

```go
func (f Static) HasStrictPrefix(prefix Static) bool {
	return len(f) > len(prefix) && prefix.Equal(f[:len(prefix)])
}
```

?

Reply:
👍

Reply:
done